rush/rush/src/actions/ChangeAction.ts

```diff
@@ -355,14 +357,31 @@ export default class ChangeAction extends CommandLineAction {
     }
   }
 
+  private _generateChangeFilePath(packageName: string): string {
+    let branch: string = undefined;
+    try {
+      branch = gitInfo().branch;
+    } catch (error) {
+      console.log('Could not automatically detect git branch name, using timestamps instead.');
+    }
+
+    const filename: string = (branch ?
```

Comment:
>filename [](start = 10, length = 8)

Whenever you are pasting strings together to make a file/path, add a TypeScript comment showing an example of what it looks like.  Something like this:

```
    const filename: string = (branch ?
      // Example: "MyBranch_20160502.json"
      this._escapeFilename(`${branch}_${this._getTimestamp()}.json`)
      // Example: "20160502.json"
      : `${this._getTimestamp()}.json`);

    // Example: "./common/changes/@scope/package/MyBranch_20160502.json"
    const filepath: string = path.join(this._rushConfiguration.commonFolder,
      'changes',
      ...packageName.split('/'),
      filename);

```

This gives a huge improvement in readability.